### ISSUE TITLE
Add an OnIdle handler to the app object

### ISSUE THREAD

**Issue by steveoc64** [fyne-io/fyne#964]
It may be useful to add an OnIdle handler to the app.

This would be fired when :
- The update queue is empty
- The event queue is empty
- A number of ticks have elapsed since the last event / update was seen

An advanced user may use this to do housekeeping in their app, but time it so it only occurs during idle periods.

Internally, we could use this to do internal housekeeping

**Comment by andydotxyz** [comment]
I think that this is part of the bigger picture lifecycle stuff that we need to look at.
More commonly needed will be "Background" API so games can pause etc.
On mobile at least that is an expectation which we can't currently match.

**Comment by andydotxyz** [comment]
Should this be added to the lifecycle ticket I have added? #1053

**Comment by Bluebugs** [comment]
I don't know if it is a lifecycle even, but if you are to add OnIdle, I would recommend to add also OnIdleEnter (Triggered only once when we start idling) and OnIdleExit (Triggered only once when we stop idling). They are easier to use right in many more case than OnIdle as they are fired on state change instead of almost continuously.

**Comment by Jacalz** [comment]
I don't think this is relevant now with the new threading model given that we run functions in between each frame?
